FirebaseMessaging/Tests/UnitTests/FIRMessagingTokenInfoTest.m

```diff
@@ -0,0 +1,199 @@
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FirebaseMessaging/Sources/Token/FIRMessagingTokenInfo.h"
+
+#import <XCTest/XCTest.h>
+
+#import "FirebaseCore/Sources/Private/FirebaseCoreInternal.h"
+#import "FirebaseMessaging/Sources/FIRMessagingUtilities.h"
+#import "FirebaseMessaging/Sources/Token/FIRMessagingAPNSInfo.h"
+#import "OCMock.h"
+
+static NSString *const kAuthorizedEntity = @"authorizedEntity";
+static NSString *const kScope = @"scope";
+static NSString *const kToken = @"eMP633ZkDYA:APA91bGfnlnbinRVE7nUwJSr_k6cuSTKectOlt66dKv1r_-"
+                                @"9Qvhy9XljAI62QPw307rgA0MaFHPnrU5sFxGZvsncRnkfuciwTUeyRpPNDZMFhNXt"
+                                @"7h1BKq9Wb2A0LAANpQefrPHVUp4p";
+static NSString *const kFirebaseAppID = @"firebaseAppID";
+static NSString *const kIID = @"eMP633ZkDYA";
+static BOOL const kAPNSSandbox = NO;
+
+@interface FIRMessagingTokenInfoTest : XCTestCase
+
+@property(nonatomic, strong) NSData *APNSDeviceToken;
+@property(nonatomic, strong) FIRMessagingTokenInfo *validTokenInfo;
+@property(nonatomic, strong) id mockOptions;
+
+@end
+
+@implementation FIRMessagingTokenInfoTest
+
+- (void)setUp {
+  [super setUp];
+
+  self.APNSDeviceToken = [@"validDeviceToken" dataUsingEncoding:NSUTF8StringEncoding];
+
+  self.mockOptions = OCMClassMock([FIROptions class]);
+  OCMStub([self.mockOptions defaultOptionsDictionary]).andReturn(@{
+    kFIRGoogleAppID : kFirebaseAppID
+  });
+
+  self.validTokenInfo =
+      [[FIRMessagingTokenInfo alloc] initWithAuthorizedEntity:kAuthorizedEntity
+                                                        scope:kScope
+                                                        token:kToken
+                                                   appVersion:FIRMessagingCurrentAppVersion()
+                                                firebaseAppID:FIRMessagingFirebaseAppID()];
+  self.validTokenInfo.APNSInfo =
+      [[FIRMessagingAPNSInfo alloc] initWithDeviceToken:self.APNSDeviceToken
+                                              isSandbox:kAPNSSandbox];
+  self.validTokenInfo.cacheTime = [NSDate date];
+
+  [[NSUserDefaults standardUserDefaults] setObject:FIRMessagingCurrentLocale()
+                                            forKey:kFIRMessagingInstanceIDUserDefaultsKeyLocale];
+}
+
+- (void)tearDown {
+  [self.mockOptions stopMocking];
+  [super tearDown];
+}
+
+- (void)testTokenInfoCreationWithInvalidArchive {
```

Comment:
I'm not sure what we are actually testing by this test? Looks like only behaviour of deprecated `[NSKeyedUnarchiver unarchiveObjectWithData:badData]` method. I would consider removing it.

Reply:
Done.